npm/private/npm_translate_lock_generate.bzl

```diff
@@ -195,39 +196,55 @@ WARNING: Issue while reading "{npmrc}". Failed to replace env in config: ${{{tok
                         npmrc = npmrc_path,
                         token = token,
                     ))
-            tokens[registry] = token
 
-        if k.find(_NPM_PKG_SCOPE_KEY) != -1:
+            if registry not in auth:
+                auth[registry] = {}
+
+            auth[registry]["bearer"] = token
+
+        # global "registry" key is special cased elsewhere
+        if k.endswith(_NPM_PKG_SCOPE_KEY):
             # @myorg:registry=https://somewhere-else.com/myorg
             # scope: @myorg
             # registry: somewhere-else.com/myorg
             scope = k.removesuffix(_NPM_PKG_SCOPE_KEY)
             registry = utils.to_registry_url(v)
             registries[scope] = registry
 
-        if k.find(_NPM_USERNAME) != -1:
+        if k == _NPM_AUTH or k.endswith(":" + _NPM_AUTH):
```

Comment:
nit: seems like the npmrc handling could be in a separate starlark source file?

Reply:
yeah; I've been meaning to move it out of here. still here from the original .bzl before the big refactor